packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -594,34 +610,126 @@ export class SDKClient {
+   */
+  async executeAllTransaction(
+    transaction: FileAppendTransaction,
+    callerName: string,
+    interactingEntity: string,
+    requestId: string,
+    shouldThrowHbarLimit: boolean,
+  ): Promise<void> {
+    const formattedRequestId = formatRequestIdMessage(requestId);
+    const transactionType = transaction.constructor.name;
     const currentDateNow = Date.now();
+    let transactionResponses: TransactionResponse[] | null = null;
+
+    // check hbar limit before executing transaction
+    if (shouldThrowHbarLimit) {
+      const shouldLimit = this.hbarLimiter.shouldLimit(currentDateNow, SDKClient.transactionMode, callerName);
+      if (shouldLimit) {
+        throw predefined.HBAR_RATE_LIMIT_EXCEEDED;
+      }
+    }
+
+    try {
+      // execute transaction
+      this.logger.info(`${formattedRequestId} Execute ${transactionType} transaction`);
+
+      transactionResponses = await transaction.executeAll(this.clientMain);
+
+      for (let transactionResponse of transactionResponses) {
+        let gasUsed: number = 0;
+        let transactionFee: number = 0;
+
+        // Since this is a loop, and it's unclear which transactionResponse might throw an error,
+        // use a try/catch/finally block to ensure the loop completes even if an error is thrown, and all the metrics are recorded.
+        try {
+          // get gasUsed and transaction fee
+          const getRecordResult = await this.executeGetTransactionRecord(transactionResponse, callerName, requestId);
+          gasUsed = getRecordResult.gasUsed;
+          transactionFee = getRecordResult.transactionFee;
+          this.logger.info(
+            `${requestId} Successfully execute ${transactionType} transaction: transactionId=${transactionResponse.transactionId}, callerName=${callerName}, transactionType=${transactionType}, status=${Status.Success}(${Status.Success._code}), cost=${getRecordResult.transactionFee} tinybars, gasUsed=${getRecordResult.gasUsed}`,
+          );
+        } catch (e: any) {
+          try {
+            const transactionRecord = await new TransactionRecordQuery()
+              .setTransactionId(transactionResponse.transactionId)
+              .setNodeAccountIds(transaction.nodeAccountIds!)
+              .setValidateReceiptStatus(false)
+              .execute(this.clientMain);
+
+            // get gasUsed and transaction fee
+            transactionFee = transactionRecord.transactionFee.toTinybars().toNumber();
+            gasUsed = transactionRecord.contractFunctionResult
+              ? transactionRecord.contractFunctionResult.gasUsed.toNumber()
+              : 0;
+          } catch (err: any) {
+            const recordQueryError = new SDKClientError(err, err.message);
+            this.logger.error(
+              recordQueryError,
+              `${formattedRequestId} Error raised during TransactionRecordQuery for ${transactionResponse.transactionId}`,
+            );
+          }
```

Comment:
Let's extract this into a separate method, nested try/catch blocks are really tough to read